Rakefile

```diff
@@ -9,4 +12,35 @@ Rake::TestTask.new do |t|
   t.verbose = true
 end
 
-task :default => :test
+Rake::TestTask.new :benchmark_tests do |t|
+  t.libs << "test"
+  t.test_files = FileList['test/**/*_benchmark.rb']
+  t.ruby_opts = ['-r./test/test_helper.rb']
+  t.verbose = true
+end
+
+task :benchmark do
+  @git = Git.init('.')
+  ref  = @git.current_branch
+
+  actual = run_benchmark_spec ref
+  master = run_benchmark_spec 'master'
+
+  @git.checkout(ref)
+
+  puts "\n\nResults ============================\n"
+  puts "------------------------------------~> (Branch) MASTER"
+  puts master
+  puts "------------------------------------\n\n"
+
+  puts "------------------------------------~> (Actual Branch) #{ref}"
+  puts actual
+  puts "------------------------------------"
+end
+
+def run_benchmark_spec(ref)
+    @git.checkout(ref)
+    response = Benchmark.realtime { Rake::Task['benchmark_tests'].invoke }
```

Comment:
So, when you switch to a branch, do you run the tests only once? Given there's warm up and all that, would it make sense to run it 2 or 3 times each branch in order to get a better average?